sct-commons/src/main/java/org/lfenergy/compas/sct/commons/scl/ied/LN0Adapter.java

```diff
@@ -237,4 +249,30 @@ private static DaTypeName getDaTypeNameForBeh() {
         daTypeNameBeh.setFc(TFCEnum.ST);
         return daTypeNameBeh;
     }
+
+    /**
+     * Update DAIs of DO InRef in all LN0 of the SCD using matching ExtRef information.
+     *
+     * @return A list of SclReport Objects that contain errors
+     */
+    public List<SclReportItem> updateDoInRef() {
+        final String daiNameToProcess = "purpose";
+        return getDOIAdapters().stream()
+                .filter(doiAdapter -> doiAdapter.getCurrentElem().isSetName()
+                        && doiAdapter.getCurrentElem().getName().contains("InRef")
+                        && doiAdapter.findDataAdapterByName(daiNameToProcess).isPresent()
+                        && !doiAdapter.getDataAdapterByName(daiNameToProcess).getCurrentElem().getVal().isEmpty())
```

Comment:
I don't think it's a good idea to ignore the possibility that the `purpose` value can be empty so we can turn it up :wink:
```suggestion
                .map(doiAdapter -> doiAdapter.getDataAdapterByName(daiNameToProcess).getCurrentElem().getVal().stream()
                                .findFirst()
                                .map(tVal -> doiAdapter.updateDaiFromExtRef(getExtRefsByDesc(tVal.getValue())))
                                .orElse(Optional.of(SclReportItem.warning(getXPath(), "The DOI %s can't be bound with an ExtRef".formatted(getXPath())))))
                .flatMap(Optional::stream)
                .toList();
```

Reply:
Ok, you're wright and Damien confirmed that in case of no-existing Val in DAI purpose, we have to create the Val attribute